pkg/server/plugin/bundlepublisher/gcpcloudstorage/gcpcloudstorage_test.go

```diff
@@ -0,0 +1,427 @@
+				Format:     "spiffe",
+			},
+			expectCode: codes.InvalidArgument,
+			expectMsg:  "configuration is missing the object name",
+		},
+		{
+			name: "no bundle format",
+			config: &Config{
+				ObjectName: "object-name",
+				BucketName: "bucket-name",
+			},
+			expectCode: codes.InvalidArgument,
+			expectMsg:  "configuration is missing the bundle format",
+		},
+		{
+			name: "client error",
+			config: &Config{
+				ServiceAccountFile: "service-account-file",
+				BucketName:         "bucket-name",
+				ObjectName:         "object-name",
+				Format:             "spiffe",
+			},
+			expectCode:   codes.Internal,
+			expectMsg:    "failed to create client: client creation error",
+			newClientErr: errors.New("client creation error"),
+		},
+		{
+			name: "invalid format",
+			config: &Config{
+				BucketName: "bucket-name",
+				ObjectName: "object-name",
+				Format:     "invalid-format",
+			},
+			expectCode: codes.InvalidArgument,
+			expectMsg:  "could not parse bundle format from configuration: unknown bundle format: \"invalid-format\"",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			options := []plugintest.Option{
+				plugintest.CaptureConfigureError(&err),
+				plugintest.CoreConfig(catalog.CoreConfig{
+					TrustDomain: spiffeid.RequireTrustDomainFromString("example.org"),
+				}),
+				plugintest.ConfigureJSON(tt.config),
+			}
+
+			newClient := func(ctx context.Context, opts ...option.ClientOption) (gcsService, error) {
+				if tt.newClientErr != nil {
+					return nil, tt.newClientErr
+				}
+				return &fakeClient{
+					clientOptions: opts,
+				}, nil
+			}
+
+			newStorageWriter := func(ctx context.Context, o *storage.ObjectHandle) io.WriteCloser {
+				return &fakeStorageWriter{}
+			}
+			p := newPlugin(newClient, newStorageWriter)
+
+			plugintest.Load(t, builtin(p), nil, options...)
+			spiretest.RequireGRPCStatusHasPrefix(t, err, tt.expectCode, tt.expectMsg)
+
+			if tt.expectMsg != "" {
+				require.Nil(t, p.config)
+				return
+			}
+
+			// Check that the plugin has the expected configuration.
+			tt.config.bundleFormat, err = bundleformat.FromString(tt.config.Format)
+			require.NoError(t, err)
+			require.Equal(t, tt.config, p.config)
+
+			client, ok := p.gcsClient.(*fakeClient)
+			require.True(t, ok)
+
+			// It's important to check that the configuration has been wired
+			// up to the gcs config, that needs to have the specified service
+			// account file.
+			require.Equal(t, []option.ClientOption{option.WithCredentialsFile(tt.config.ServiceAccountFile)}, client.clientOptions)
+		})
+	}
+}
+
+func TestPublishBundle(t *testing.T) {
+	testBundle := getTestBundle(t)
+
+	for _, tt := range []struct {
+		name string
+
+		newClientErr error
+		expectCode   codes.Code
+		expectMsg    string
+		config       *Config
+		bundle       *types.Bundle
+		writeErr     error
+		closeErr     error
+	}{
+		{
+			name:   "success",
+			bundle: testBundle,
+			config: &Config{
```

Comment:
since config is  pretty much the same except test that validates that plugin was never configured, 
what do you think about adding a bool to tt, something like `noConfigure bool` and just no run configure when that is true, and make this config always default option?